lib/handler/file.c

```diff
@@ -110,6 +127,57 @@ static void do_proceed(h2o_generator_t *_self, h2o_req_t *req)
         do_close(&self->super, req);
 }
 
+static void do_multirange_proceed(h2o_generator_t *_self, h2o_req_t *req) {
+    struct st_h2o_sendfile_generator_t *self = (void *)_self;
+    size_t rlen, *range_cur = self->ranged.range_infos + 2 * self->ranged.finished_range, used_buf = 0;
+    ssize_t rret, vecarrsize;
+    h2o_iovec_t vec, *vecarr;
+    int is_finished;
+
+    if (self->ranged.current_range == self->ranged.finished_range) {
+        size_t range_end = *range_cur + *(range_cur + 1) - 1;
+        if (H2O_LIKELY(self->ranged.current_range != 0))
+            used_buf = sprintf(self->buf, "\r\n--%s\r\nContent-Type: %s\r\nContent-Range: bytes %zd-%zd/%zd\r\n\r\n",
+                               self->ranged.boundary.base, self->ranged.mimetype.base, *range_cur, range_end,
+                               self->ranged.filesize);
+        else
+            used_buf = sprintf(self->buf, "--%s\r\nContent-Type: %s\r\nContent-Range: bytes %zd-%zd/%zd\r\n\r\n",
+                               self->ranged.boundary.base, self->ranged.mimetype.base, *range_cur, range_end,
+                               self->ranged.filesize);            
+        self->ranged.current_range ++;
+        lseek(self->fd, *range_cur, SEEK_SET);
```

Comment:
Please close the response if `lseek` fails, in the same way as when `read` fails.